Bring back fixes from 0.9.0-SHINE build

sukibaby opened the pull request:
`beta` has an awful compounding drift problem and none of the beta builds have been preferred over 0.9.0-SHINE for some months. Many people are still playing on 0.9.0-SHINE. This PR will bring back the majority of code which differs between the beta and 0.9.0-SHINE.

One of the RSMB PR's is needed in addition to this.

sukibaby committed:
Truly be DPI aware (Windows)

This commit keeps 4K screens working correctly.

Revert "Windows: Mark executable as DPI aware"

This reverts commit daad1a23337dcf1e7ed6e76880be171640c9f2ce.

teejusb commented:
> Reverts to the BitmapText::AddAttribute

I'm going to disagree on reverting this. I'll review the code after you remove this from the list. Also don't name the commit
`bring back stable code from SHINE build` because it doesnt give any information about the history and actually label them as the individual commits being reverted.

sukibaby committed:
Rewrite of RageSoundDriverNull::GetPosition

Handles casting of different variable types in a more clear way.

sukibaby committed:
Better optimize RageSoundDriver_Generic_Software

1)   Define a macro for all the different places '1000000` is used

2)   Optimize the ASSERT_M for cases where it doesn't fail.  Actually,  GetSampleRate() is supposed to return a hardcoded value of 44100 when it's called, so I'm not sure that any assertion is needed here. If it ends up being 0 somehow then it's going to crash anyway.

3)  Use std::max instead of an if loop, the compiler can optimize that better.

4)  Remove a conditional check in GetHardwareFrame that is only really needed in debugging.

sukibaby committed:
Use old assignment operator from RageSoundPosMap

as it was in 0.9.0-SHINE

sukibaby commented:
> > Reverts to the BitmapText::AddAttribute
> 
> I'm going to disagree on reverting this. I'll review the code after you remove this from the list. Also don't name the commit `bring back stable code from SHINE build` because it doesnt give any information about the history and actually label them as the individual commits being reverted.

I have broken that commit out to be more descriptive, as shown below


The commits from "bring back stable code from SHINE build" has been broken out like so:


## Better optimize RageSoundDriver_Generic_Software
1)   Define a macro for all the different places '1000000` is used

2)   Optimize the ASSERT_M for cases where it doesn't fail.  Actually,  GetSampleRate() is supposed to return a hardcoded value of 44100 when it's called, so I'm not sure that any assertion is needed here. If it ends up being 0 somehow then it's going to crash anyway.

3)  Use std::max instead of an if loop, the compiler can optimize that better.

4)  Remove a conditional check in GetHardwareFrame that is only really needed in debugging.

## Rewrite of RageSoundDriverNull::GetPosition
Handles casting of different variable types in a more clear way.

## RageSoundMixBuffer
### Please choose between the fixed buffer size code in #543, or the [vector-based RSMB from 12-20's test build](https://github.com/itgmania/itgmania/pull/568)

## Use old assignment operator from RageSoundPosMap
as it was in 0.9.0-SHINE

teejusb commented:
Can you link the commits that you're reverting as well

sukibaby commented:
> Can you link the commits that you're reverting as well

https://github.com/itgmania/itgmania/commit/daad1a23337dcf1e7ed6e76880be171640c9f2ce

https://github.com/itgmania/itgmania/pull/318/commits/3361859e8f61366185c33e3a5619712af0a66fbf

sukibaby commented:
This wasn't part of the jan 6 build so I am closing it.
